// @bexis2/bexis2-core-ui/dist/models/Enums.js
export var positionType;
(function (positionType) {
    positionType["start"] = "start";
    positionType["end"] = "end";
    positionType["center"] = "center";
})(positionType || (positionType = {}));
export var pageContentLayoutType;
(function (pageContentLayoutType) {
    pageContentLayoutType["full"] = "full";
    pageContentLayoutType["center"] = "center";
})(pageContentLayoutType || (pageContentLayoutType = {}));
export var decimalCharacterType;
(function (decimalCharacterType) {
    decimalCharacterType[decimalCharacterType["point"] = 0] = "point";
    decimalCharacterType[decimalCharacterType["comma"] = 1] = "comma";
})(decimalCharacterType || (decimalCharacterType = {}));
export var orientationType;
(function (orientationType) {
    orientationType[orientationType["columnwise"] = 0] = "columnwise";
    orientationType[orientationType["rowwise"] = 1] = "rowwise";
})(orientationType || (orientationType = {}));
export var textSeperatorType;
(function (textSeperatorType) {
    textSeperatorType[textSeperatorType["tab"] = 9] = "tab";
    textSeperatorType[textSeperatorType["comma"] = 44] = "comma";
    textSeperatorType[textSeperatorType["semicolon"] = 59] = "semicolon";
    textSeperatorType[textSeperatorType["space"] = 32] = "space";
})(textSeperatorType || (textSeperatorType = {}));
export var textMarkerType;
(function (textMarkerType) {
    textMarkerType[textMarkerType["quotes"] = 0] = "quotes";
    textMarkerType[textMarkerType["doubleQuotes"] = 1] = "doubleQuotes";
})(textMarkerType || (textMarkerType = {}));
//types for the notification styling
export var notificationType;
(function (notificationType) {
    notificationType[notificationType["success"] = 0] = "success";
    notificationType[notificationType["warning"] = 1] = "warning";
    notificationType[notificationType["error"] = 2] = "error";
    notificationType[notificationType["surface"] = 3] = "surface";
})(notificationType || (notificationType = {}));
